Improve Volume snapshotting guide (#8332)
kyma-project/kyma: docs/backup/03-01-new-backup-strategy.md

For details about VolumeSnapshots, see [this](https://kubernetes.io/docs/concepts/storage/volume-snapshots/) document.

> Follow the instructions below to enable this feature for various providers.

Reviewer: ```suggestion
Follow the instructions to enable this feature for various providers.
```

### AKS

Minimum Kubernetes version supported is 1.17.

1. Install CSI driver following [this documentation](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/install-csi-driver-master.md).

2. Follow [this example](https://github.com/kubernetes-sigs/azuredisk-csi-driver/tree/master/deploy/example/snapshot) to see how you can create a VolumeSnapshot.

### GKE

Minimum Kubernetes version supported is 1.14.

1. Enable the required feature gate on the cluster following [this](https://cloud.google.com/kubernetes-engine/docs/how-to/gce-pd-csi-driver) document.

2. Check out [this repository](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver) for the details on how to use VolumeSnapshots on GKE.

### Gardener

#### GCP

> Gardener GCP uses CSI drivers by default as of Kubernetes 1.18, and it supports Volume snapshotting out of the box.

Reviewer: ```suggestion
Gardener GCP uses CSI drivers by default as of Kubernetes 1.18, and it supports taking volume snapshots out of the box.
```

> Create a VolumeSnapshotClass:

Reviewer: ```suggestion
1. Create a VolumeSnapshotClass:
```

```yaml
apiVersion: snapshot.storage.k8s.io/v1beta1
kind: VolumeSnapshotClass
metadata:
  annotations:
    snapshot.storage.kubernetes.io/is-default-class: "true"
  name: snapshot-class
driver: pd.csi.storage.gke.io
deletionPolicy: Delete
```

> Create a VolumeSnapshot:

Reviewer: ```suggestion
2. Create a VolumeSnapshot:
```

```yaml
apiVersion: snapshot.storage.k8s.io/v1beta1
kind: VolumeSnapshot
metadata:
  name: snapshot
spec:
  source:
    persistentVolumeClaimName: <PVC_NAME>
```

> Wait until `READYTOUSE` field gets `true` to verify that snapshotting gets succeeded:

Reviewer: ```suggestion
3. Wait until the **READYTOUSE** field has the `true` status to verify that the snapshot was taken successfully:
```

```bash
kubectl get volumesnapshot -w
```

#### AWS

> Gardener AWS uses CSI drivers by default as of Kubernetes 1.18, and it supports Volume snapshotting out of the box.

Reviewer: ```suggestion
Gardener AWS uses CSI drivers by default as of Kubernetes 1.18, and it supports taking volume snapshots out of the box.
```

> Create a VolumeSnapshotClass:

Reviewer: ```suggestion
1. Create a VolumeSnapshotClass:
```

```yaml
apiVersion: snapshot.storage.k8s.io/v1beta1
kind: VolumeSnapshotClass
metadata:
  annotations:
    snapshot.storage.kubernetes.io/is-default-class: "true"
  name: snapshot-class
driver: ebs.csi.aws.com
deletionPolicy: Delete
```

> Create a VolumeSnapshot:

Reviewer: ```suggestion
2. Create a VolumeSnapshot:
```

```yaml
apiVersion: snapshot.storage.k8s.io/v1beta1
kind: VolumeSnapshot
metadata:
  name: snapshot
spec:
  source:
    persistentVolumeClaimName: <PVC_NAME>
```

> Wait until `READYTOUSE` field gets `true` to verify that snapshotting gets succeeded:

Reviewer: ```suggestion
3. Wait until the **READYTOUSE** field receives the `true` status to verify that the snapshot was taken successfully:
```
